classes/webservice/WebserviceOutputBuilder.php

```diff
@@ -660,12 +660,14 @@ protected function renderAssociations($object, $depth, $associations, $ws_params
                 $output_details = '';
                 foreach ($objects_assoc as $object_assoc) {
                     if ($depth == 0 || $class_name === null) {
-                        $value = null;
-                        if (!empty($object_assoc)) {
-                            $value = $object_assoc;
+                        if (empty($object_assoc)) {
+                            continue;
                         }
+                        $value = $object_assoc;
                         if (empty($fields_assoc)) {
-                            $fields_assoc = [['id' => $value['id']]];
+                            if (!empty($value['id'])) {
+                                $fields_assoc = [['id' => $value['id']]];
+                            }
```

Comment:
Can you please use `AND` operator instead a new `IF` statement ? (see the line just bellow)